x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/xcontent/XContentUtils.java

```diff
@@ -102,26 +105,40 @@ public static void addAuthorizationInfo(final XContentBuilder builder, final Map
             return;
         }
         builder.startObject("authorization");
-        switch (authenticationSubject.getType()) {
-            case USER -> builder.array(User.Fields.ROLES.getPreferredName(), authenticationSubject.getUser().roles());
+        addSubjectInfo(builder, authenticationSubject);
+        builder.endObject();
+    }
+
+    private static void addSubjectInfo(XContentBuilder builder, Subject subject) throws IOException {
+        switch (subject.getType()) {
+            case USER -> builder.array(User.Fields.ROLES.getPreferredName(), subject.getUser().roles());
             case API_KEY -> {
-                builder.startObject("api_key");
-                Map<String, Object> metadata = authenticationSubject.getMetadata();
-                builder.field("id", metadata.get(AuthenticationField.API_KEY_ID_KEY));
-                Object name = metadata.get(AuthenticationField.API_KEY_NAME_KEY);
-                if (name instanceof String) {
-                    builder.field("name", name);
-                }
-                builder.endObject();
+                addApiKeyInfo(builder, subject);
             }
-            case SERVICE_ACCOUNT -> builder.field("service_account", authenticationSubject.getUser().principal());
+            case SERVICE_ACCOUNT -> builder.field("service_account", subject.getUser().principal());
             case CROSS_CLUSTER_ACCESS -> {
-                // TODO handle cross cluster access authentication
-                final String message = "cross cluster access authentication is not yet supported";
-                assert false : message;
-                throw new UnsupportedOperationException(message);
+                builder.startObject("cross_cluster_access");
+                {
+                    addApiKeyInfo(builder, subject);
+                    builder.startObject("remote_authorization");
```

Comment:
This field is named `remote_authorization` to be both different from and consistent with the outer `authorization` field. I am open to suggestions.

Reply:
Ya I think `remote_authorization` works. Only alternative I can think of is `authorization_from_remote` but I don't have a preference either way.